### PR TITLE
Parallelize CI builds for tests

### PR DESCRIPTION
**Pull request by jamilbk**
Refs firezone/product#399

@firezone/engineering

This PR separates the `test` steps from the `build` steps.
- All PRs will only run the `test` steps by default, which are parallelized across ephemeral GH public runners
- The `build` steps will now only run on `master` and any branch with a name beginning with `build/`.

This should speed up day-to-day CI runs and get us closer to a fully-parallelizable e2e pipeline.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  push:
+
+jobs:
+  lint-docs:
+    runs-on: macos-11
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14.19.3'
```
Comment (princemaple) [inline .github/workflows/build.yml]: 16?

Comment (jamilbk) [reply .github/workflows/build.yml]: Yeah looks like that's sitting in `0.5.0-stage` waiting for merge. I went ahead and updated it here anyway as well.

**Review by princemaple — APPROVED** [review]
Fast builds are the best builds.

**Comment by princemaple** [comment]
Do you think it's a good idea to build one version for running functional test?

**Comment by jamilbk** [comment]
> Do you think it's a good idea to build one version for running functional test?

yep let me open a pr for that!

**Pull request merged**